### PR TITLE
BBS-39 port ding_serendipity_ting_entity to SAL

### PR DESCRIPTION
**Pull request by danquah**

### CODE REVIEW

File: modules/p2/ding_serendipity_ting_entity/ding_serendipity_ting_entity.module [hunk 1]
```diff
@@ -141,24 +145,26 @@ function ding_serendipity_ting_entity_ting_object_subject_serendipity_add($conte
...
+  }, $subjects);
 
-    foreach ($results as & $item) {
-      $item['info'] = array(
-        'description' => t('This material has been selected based on the subjects of the material you are viewing.'),
-      );
-    }
+  $query = ting_start_query()->addFieldFilters($field_filters);
+  $results = ding_serendipity_do_search($query);
+  ding_serendipity_exclude($results, $context['ting_object_id']);
```
Comment (xendk) [inline modules/p2/ding_serendipity_ting_entity/ding_serendipity_ting_entity.module:162]: Wouldn't an extra `AND NOT rec.id =` be smarter than this?

File: modules/p2/ding_serendipity_ting_entity/ding_serendipity_ting_entity.module [hunk 2]
```diff
@@ -103,20 +107,20 @@ function ding_serendipity_ting_entity_ting_object_author_serendipity_add($contex
     return $results;
   }
 
-  foreach ($creators as & $creator) {
-    $creator = 'facet.creator="' . $creator . '"';
+  // Find related content from any of the authors.
+  $field_filters = [];
+  foreach ($creators as $creator) {
+    $field_filters[] = new TingSearchFieldFilter(TingSearchCommonFields::AUTHOR, $creator, '=', BooleanStatementInterface::OP_OR);
```
Comment (xendk) [inline modules/p2/ding_serendipity_ting_entity/ding_serendipity_ting_entity.module:113]: The or statement is on the filter? How does that work with mixed AND and ORs?

Wouldn't it be smarter with AND and OR objects that contain field filters?

And why aren't we duplicating the $query->condition(...) pattern of EntityFieldQuery?

Comment (danquah) [reply modules/p2/ding_serendipity_ting_entity/ding_serendipity_ting_entity.module:113]: I'll do that in a follow up ticket!

File: modules/p2/ding_serendipity_ting_entity/ding_serendipity_ting_entity.module (hunk 1 above)
Comment (danquah) [reply modules/p2/ding_serendipity_ting_entity/ding_serendipity_ting_entity.module:162]: I'll add a note and fix it in the future.

**Pull request merged**